File: examples/online-store/templates-eta/_gen_context.js
import { readFileSync } from "node:fs";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";

const __dirname = dirname(fileURLToPath(import.meta.url));

export function getContexts(parentContext) {
  const modelPath = join(__dirname, "..", "models", "model.json");
  const model = JSON.parse(readFileSync(modelPath, "utf-8"));

  const entities = model.entities;
  const entityMap = Object.fromEntries(entities.map((e) => [e.name, e]));

  for (const entity of entities) {
    entity.all_entities = entities;
    entity.entity_map = entityMap;
  }

  return entities.map((entity) => ({ entity }));
}

File: examples/online-store/templates-jinja2/_gen_context.py
import json
from pathlib import Path


def get_contexts(parent_context):
    model_path = Path(__file__).parent.parent / "models" / "model.json"
    with open(model_path) as f:
        model = json.load(f)

    entities = model["entities"]

    # Build a lookup so templates can find related entities by name
    entity_map = {e["name"]: e for e in entities}

    # Attach the full entity list and lookup to each entity
    for entity in entities:
        entity["all_entities"] = entities
        entity["entity_map"] = entity_map

    return [{"entity": entity} for entity in entities]
